src/dmd/parse.d

```diff
@@ -7914,112 +7914,31 @@ final class Parser(AST) : Lexer
-                        case TOK.functionString:
-                        case TOK.prettyFunction:
-                        case TOK.wchar_:
-                        case TOK.dchar_:
-                        case TOK.bool_:
-                        case TOK.char_:
-                        case TOK.int8:
-                        case TOK.uns8:
-                        case TOK.int16:
-                        case TOK.uns16:
-                        case TOK.int32:
-                        case TOK.uns32:
-                        case TOK.int64:
-                        case TOK.uns64:
-                        case TOK.int128:
-                        case TOK.uns128:
-                        case TOK.float32:
-                        case TOK.float64:
-                        case TOK.float80:
-                        case TOK.imaginary32:
-                        case TOK.imaginary64:
-                        case TOK.imaginary80:
-                        case TOK.complex32:
-                        case TOK.complex64:
-                        case TOK.complex80:
-                        case TOK.void_:
-                            {
-                                // (type) una_exp
-                                nextToken();
-                                auto t = parseType();
-                                check(TOK.rightParentheses);
+                        if (tk.value == TOK.dot) {
+                            // (type) una_exp
+                            nextToken();
+                            auto t = parseType();
+                            check(TOK.rightParentheses);
 
-                                // if .identifier
-                                // or .identifier!( ... )
-                                if (token.value == TOK.dot)
-                                {
-                                    if (peekNext() != TOK.identifier && peekNext() != TOK.new_)
-                                    {
-                                        error("identifier or new keyword expected following `(...)`.");
-                                        return null;
-                                    }
-                                    e = new AST.TypeExp(loc, t);
-                                    e = parsePostExp(e);
-                                }
-                                else
+                            // if .identifier
+                            // or .identifier!( ... )
+                            if (token.value == TOK.dot)
+                            {
+                                if (peekNext() != TOK.identifier && peekNext() != TOK.new_)
                                 {
-                                    e = parseUnaryExp();
-                                    e = new AST.CastExp(loc, e, t);
-                                    error("C style cast illegal, use `%s`", e.toChars());
+                                    error("identifier or new keyword expected following `(...)`.");
```

Comment:
This does not appear to be covered by any test.  Please add coverage.  You can add a test to the fail_compilation directory to verify compile errors.